regNo_167249_ stack

pphantomfury opened the pull request:
The code provided  in a simple stack written in python code. The code enables administrator to edit or add clarifications about the type of illness each patient is suffering from. The changes can be undone, modified  or completely removed from the respective patient, using their names or patient ID.

github-classroom[bot] committed:
GitHub Classroom Feedback

github-classroom[bot] committed:
Setting up GitHub Classroom Feedback

github-classroom[bot] committed:
add online IDE url; add deadline

pphantomfury committed:
first commit

pphantomfury committed:
first commit

pphantomfury commented:
the code has been reviewed by 189720

pphantomfury committed:
last commit

NoelaJepchumba reviewed (approved):
Code looks good to me. The structure is clear, and I like how the undo logic is handled. It runs as expected, and I didn’t notice any issues.

pphantomfury committed:
first commit

pphantomfury committed:
FINAL commit

pphantomfury committed:
commit changes in undo_stack.py

pphantomfury committed:
last commit

pphantomfury committed:
last commit

The initial code had problems for updating patients using the stack(Last in, Last out) stack format. The new changes provided ensures that that functionality is possible. All updates are done from the top element of the stack and an element can only be deleted from the top element in the stack.
The when the maximum size is reached an element cannot be added to the already full stack, although updating and deleting processes are still possible .

pphantomfury committed:
Lat commit message

I made final commit message since , i realized that I was not supposed to perform all (CRUD) functionalities at the same time. I removed the previous patient class and focused on the undo functionality of the stack alone.

Kirwa16 committed:
first commit
